### PR TITLE
[SYCL][Doc] Require put link to test PR in commit message

### PR DESCRIPTION
**Pull request by vladimirlaz**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,6 +62,12 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
 - For any DPC++-related commit, the `[SYCL]` tag should be present in the
   commit message title. To a reasonable extent, additional tags can be used
   to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.
+- For product changes which require modification of E2E tests
```
Comment (pvchupin) [inline CONTRIBUTING.md:65]: It probably makes sense to replace E2E here with more generic: "tests, outside of this repository"

**Review by pvchupin — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -62,6 +62,12 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
...
+  (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain))
+  the commit message should contain link to corresponding test PR, e.g.: E2E
```
Comment (romanovvlad) [inline CONTRIBUTING.md:67]: Could you please clarify if the intent here is that a link should be in final commit message or in the first PR message or both?

Comment (pvchupin) [reply CONTRIBUTING.md:67]: Thinking more about that... Should we also require that PR to have clear message which will be used as final merge commit message when many commits squashed?
If we do this link to test should be in PR.

File: CONTRIBUTING.md (hunk 1 above)
Comment (vladimirlaz) [reply CONTRIBUTING.md:65]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (vladimirlaz) [reply CONTRIBUTING.md:67]: done

**Review by romanovvlad — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -62,6 +62,14 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
 - For any DPC++-related commit, the `[SYCL]` tag should be present in the
   commit message title. To a reasonable extent, additional tags can be used
   to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.
+- For product changes which require modification in tests outside current repo
```
Comment (bader) [inline CONTRIBUTING.md:65]:
```suggestion
- For product changes which require modification in tests outside of the current repository
```
?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -62,6 +62,14 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
...
+  (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain))
```
Comment (bader) [inline CONTRIBUTING.md:66]:
```suggestion
  (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain)),
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -62,6 +62,14 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
...
+  the commit message should contain link to corresponding test PR, e.g.: E2E
```
Comment (bader) [inline CONTRIBUTING.md:67]:
```suggestion
  the commit message should contain the link to corresponding test PR, e.g.: E2E
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -62,6 +62,14 @@ see [ABI Policy Guide](sycl/doc/ABIPolicyGuide.md) for more information.
...
+  test changes are available under intel/llvm-test-suite#88 or SYCL
+  conformance test changes are available under KhronosGroup/SYCL-CTS#65 (see
+  [Autolinked references and URLs](https://docs.github.com/en/free-pro-team/github/writing-on-github/autolinked-references-and-urls)
+  for more details). The same message should be present both in commit
+  message and PR description.
```
Comment (bader) [inline CONTRIBUTING.md:72]:
```suggestion
  message and in PR description.
```

**Review by bader — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**